lib/fluent/plugin/filter_throttle.rb

```diff
@@ -0,0 +1,137 @@
+require 'fluent/filter'
+
+module Fluent
+  class ThrottleFilter < Filter
+    Fluent::Plugin.register_filter('throttle', self)
+
+    config_param :group_key, :string, :default => 'kubernetes.container_name'
+    config_param :group_bucket_period_s, :integer, :default => 60
+    config_param :group_bucket_limit, :integer, :default => 6000
+    config_param :group_reset_rate_s, :integer, :default => nil
+    config_param :warning_hz, :float, :default => 0.1
+
+    Group = Struct.new(
+      :rate_count,
+      :rate_last_reset,
+      :aprox_rate,
+      :bucket_count,
+      :bucket_last_reset,
+      :last_warning)
+
+    def configure(conf)
+      super
+
+      @group_key_path = group_key.split(".")
+
+      raise "group_bucket_period_s must be > 0" \
+        unless @group_bucket_period_s > 0
+
+      raise "group_bucket_limit must be > 0" \
+        unless @group_bucket_limit > 0
+
+      @group_rate_limit = (@group_bucket_limit / @group_bucket_period_s)
+
+      @group_reset_rate_s = @group_rate_limit \
+        if @group_reset_rate_s == nil
+
+      raise "group_reset_rate_s must be >= -1" \
+        unless @group_reset_rate_s >= -1
+      raise "group_reset_rate_s must be <= group_bucket_limit / group_bucket_period_s" \
+        unless @group_reset_rate_s <= @group_rate_limit
+
+      @warning_delay = (1.0 / @warning_hz)
```

Comment:
Should we check for non zero of @warning_hz or directly expose warning_delay?

Reply:
as discussed offline, will change the config to a delay in s directly.